### ISSUE TITLE
Amethyst 0.16 stops tiling any windows on Ventura

### ISSUE THREAD

**Issue by saurabhsharan** [ianyh/Amethyst#1341]
**Describe the bug**
I'm not sure what changed but Amethyst was working great until yesterday when it stopped tiling any windows. I've tried rebooting my Mac and reinstalling Amethyst but I can't seem to get tiling to work anymore.

**Applications:**
It doesn't seem to work for any apps, but I am trying it with Chrome, iTerm2, and VSCode.

**To Reproduce**
Launch/re-launch Amethyst and the tiling doesn't happen. Nor does the tiling happen when changing the layout.

**Expected behavior**
Amethyst should tile the windows automatically or after manually triggering a layout.

**Screenshots**
N/A

**Versions:**
 - macOS: 13.0
 - Amethyst: 0.16.0 (94)
 
**Debug Info**
```
mouse-follows-focus: 0
follow-space-thrown-windows: 1
screen-padding-right: 0
floating: (
)
floating-is-blacklist: 1
restore-layouts-on-launch: 1
screen-padding-bottom: 0
new-windows-to-main: 1
window-margins: 0
debug-layout-info: 0
use-canary-build: 0
screen-padding-left: 0
ignore-menu-bar: 0
screen-padding-top: 0
window-minimum-width: 0
mod1: (
    option,
    shift
)
window-resize-step: 5
window-minimum-height: 0
mouse-swaps-windows: 1
enables-layout-hud-on-space-change: 1
layouts: (
    "two-pane",
    tall,
    column,
    "middle-wide"
)
enables-layout-hud: 1
mod2: (
    option,
    shift,
    control
)
focus-follows-mouse: 0
float-small-windows: 1
window-margin-size: 12

Manageable applications:
	talagent (com.apple.talagent)
	Finder (com.apple.finder)
	Single Sign-On (com.apple.AppSSOAgent)
	1Password (com.1password.1password)
	Google Chrome (com.google.Chrome)
	Code (com.microsoft.VSCode)
	iTerm2 (com.googlecode.iterm2)
```

**Additional context**

**Comment by qmacro** [comment]
I experienced this same behaviour today. macOS Ventura 13.0.1 (22A400). Amethyst and Chrome are in my Login Items "Open at Login". On OS startup, Chrome windows were not tiled, although other apps' windows were.

I found that quitting Chrome and restarting it enabled Amethyst to tile it as normal after that. 

After starting up Kitty (terminal) and opening up a couple of windows I noticed that they weren't tiled properly either, but switching to another virtual desktop ("space"?) and back again saw them tiled correctly when switched back. This behaviour continued too (new Kitty windows weren't tiled correctly until doing a virtual desktop switch-and-back).

**Comment by davidwincent** [comment]
The whole window manager experience is broken for me after upgrade to Ventura.

Issues:

- Switching layouts no longer auto arranges the windows. Possible workarounds involve switching to another workspace and back
- Any changing of window layout / sizes involves a weird animation which I have found no way to disable
- Some windows pinned as floating (Slack) never leaves foreground, even if focusing on another window

| Component | Version |
| --- | --- |
| macOS |  Ventura 13.0.1 |
| Amethyst |  0.16.0 (94) |

Amethyst installed using `brew`

**Comment by zachmeador** [comment]
I've held off on upgrading to Ventura solely because of this issue because Amethyst is a key part of my working environment. Can we have some clarification on the main GH page, and website, if Ventura is actually supported or not at this point? 

I see some users are able to use it, but others are not? I also see a pull request for 16.1, but there doesn't appear to be any development happening? 

Any clarification of any kind would be appreciated! Would be very sad to see this project abandoned.

**Comment by davidwincent** [comment]
> The whole window manager experience is broken for me after upgrade to Ventura.
> 
> Issues:
> 
> * Switching layouts no longer auto arranges the windows. Possible workarounds involve switching to another workspace and back
> * Any changing of window layout / sizes involves a weird animation which I have found no way to disable
> * Some windows pinned as floating (Slack) never leaves foreground, even if focusing on another window
> 
> Component	Version
> macOS	Ventura 13.0.1
> Amethyst	0.16.0 (94)
> Amethyst installed using `brew`

https://user-images.githubusercontent.com/3088566/208240007-d6b781fb-6d46-495c-8ec9-8b3f8a1336cd.mov



Attaching video of experience with Amethyst and macOS Ventura,
Upgrading macOS to 13.1 did not change anything.

**Comment by saurabhsharan** [comment]
After spending a couple hours diving into the codebase, it seems like one issue is on this line:

https://github.com/ianyh/Amethyst/blob/c36a40224fe28fc081a89800212b855b32a69620/Amethyst/Managers/Windows.swift#L52

It seems like `isFloating` is always `true` for every window (even non-floating windows), causing the list of active windows to be empty.

As a quick test to verify, if I change this line where we filter the active windows for Amethyst to tile:

https://github.com/ianyh/Amethyst/blob/c36a40224fe28fc081a89800212b855b32a69620/Amethyst/Managers/Windows.swift#L54

to not check for floating window (i.e. delete the `&& !isFloating` at the end), then Amethyst seems to mostly work fine on Ventura 13.0 again (although obviously will not work correctly if you have any floating windows). So it seems like Ventura may have broken the way of determining whether a window is floating or not.

/cc @ianyh in case you have seen something like this before

**Comment by saurabhsharan** [comment]
After some more fiddling around I've been able to mostly get Amethyst working on Ventura again. The issue for me was related to determining whether a window is floating (and thus whether it should be tiled), and more specifically seems related to the "Automatically float applications listed" preference in the Floating settings pane. In my case, somehow that preference field had gotten corrupt on-disk and Amethyst was reading an incorrect value. To reset it I ran this command (after quitting Amethyst):

`defaults delete com.amethyst.Amethyst floating-is-blacklist`

Then launched Amethyst and made sure that field was set to "Automatically float applications listed" in the Floating settings pane. Once I did that Amethyst started working again. (Note that I am still on Ventura 13.0 and haven't tried it on 13.1 yet.)

**Comment by davidwincent** [comment]
> After some more fiddling around I've been able to mostly get Amethyst working on Ventura again. The issue for me was related to determining whether a window is floating (and thus whether it should be tiled), and more specifically seems related to the "Automatically float applications listed" preference in the Floating settings pane. In my case, somehow that preference field had gotten corrupt on-disk and Amethyst was reading an incorrect value. To reset it I ran this command (after quitting Amethyst):
> 
> `defaults delete com.amethyst.Amethyst floating-is-blacklist`
> 
> Then launched Amethyst and made sure that field was set to "Automatically float applications listed" in the Floating settings pane. Once I did that Amethyst started working again. (Note that I am still on Ventura 13.0 and haven't tried it on 13.1 yet.)

Tried above command and removing everything in "Automatically float applications listed". Still need to click on windows multiple times before they arrange after switching layout.

**Comment by sspaeti** [comment]
I have the same behavior as shown [above](https://github.com/ianyh/Amethyst/issues/1341#issuecomment-1356223149), but already long before the Ventura upgrade. But as discussed in other issues, it seems to be the browser only (Brave in my case, but as reading the others, Chrome behaviors are the same).

If I use apps other than Brave, it works like a charm. Unfortunately, the browser is the one I use most of the time in combination with Amethysts.

**Comment by glottisfaun0000** [comment]
i've been having a tough-to-reproduce issue where windows brought back from cmd-H hiding, minimizing, or new windows won't tile and just sit on top of other windows until I relaunch amethyst. may be something else altogether, but i thought it coincided with ventura.

**Comment by JonathanP-Git** [comment]
> I have the same behavior as shown [above](https://github.com/ianyh/Amethyst/issues/1341#issuecomment-1356223149), but already long before the Ventura upgrade. But as discussed in other issues, it seems to be the browser only (Brave in my case, but as reading the others, Chrome behaviors are the same).
> 
> If I use apps other than Brave, it works like a charm. Unfortunately, the browser is the one I use most of the time in combination with Amethysts.

I've also experienced this problem, particularly when using Chrome on Monterey and Ventura. However, I've found that using the "Force windows to be reevaluated" shortcut can be an effective workaround.

**Comment by simgt** [comment]
> Tried above command and removing everything in "Automatically float applications listed". Still need to click on windows multiple times before they arrange after switching layout.

Same issue here, it seems really due to a window being mistaken for a floating one. It occurs too often to be practical, sadly.
Changing focus a couple times makes the tiling work again, but forcing windows reevaluation does not.

EDIT: ~It looks like the problem has been solved for me by deleting all the configuration.~ Actually no.

```bash
# Quit Amethyst then:
defaults delete com.amethyst.Amethyst
# Check that everything is gone:
defaults find com.amethyst.Amethyst
```

**Comment by ianyh** [comment]
There's actually a bunch of issues in this thread.
* @davidwincent's issue—thanks for the screen recording!—is a duplicate of #1257. A fix is being worked on in #1276.
* The issue that is resolved with `defaults` fiddling and the `floating-is-blacklist` key should be (mostly) addressed by #1363, though there might be some extra intervention needed. Maybe I need a common issues page or something.
* The issue with hiding and restoring apps not correctly tracking windows is (I think) tracked in #1191.
* @saurabhsharan's issue with floating is resolved in #1362.

**Comment by Morgy93** [comment]
Not sure if it's related, but for me it breaks the moment I change `Automatically float applications listed` to `Automatically float all applications except those listed` and restart Amethyst.

If not related, I'll open up another issue.

**Comment by simgt** [comment]
> Not sure if it's related, but for me it breaks the moment I change `Automatically float applications listed` to `Automatically float all applications except those listed` and restart Amethyst.

I have `Automatically float applications listed` but still experience the issue on the latest release. I still can't really tell when it occurs.

**Comment by saurabhsharan** [comment]
Closing since this issue is fixed for me as of Amethyst 0.17.0.
